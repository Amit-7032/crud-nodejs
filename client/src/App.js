import React, { useState, useEffect } from "react";
import List from "./components/List";
import axios from "axios";
import { baseURL } from "./utils/constant";
import "./index.css";
import CrudOperation from "./components/CrudOperation";

function App() {
  const [inputData, setInputData] = useState("");
  const [tasks, setTasks] = useState([]);
  const [updateUI, setUpdateUI] = useState(false);
  const [updateId, setUpdateId] = useState(null);

  useEffect(() => {
    axios.get(`${baseURL}/show`).then((res) => {
      setTasks(res.data);
    });
  }, [updateUI]);

  const addTask = () => {
    axios.post(`${baseURL}/add`, { task: inputData }).then((res) => {
      setInputData("");
      setUpdateUI((prevState) => !prevState);
    });
  };

  const updateMode = (id, text) => {
    setInputData(text);
    setUpdateId(id);
  };

  const updateTask = () => {
    axios
      .put(`${baseURL}/update/${updateId}`, { task: inputData })
      .then((res) => {
        setUpdateUI((prevState) => !prevState);
        setUpdateId(null);
        setInputData("");
      });
  };

  return (
    <>
      <div className="container">
        <h1 className="title">CRUD Operations</h1>
        <div className="input-container">
          <input
            type="text"
            value={inputData}
            onChange={(e) => setInputData(e.target.value)}
            style={{ border: "2px solid #0C090A" }}
          />
          <button
            type="submit"
            onClick={updateId ? updateTask : addTask}
            style={{ border: "2px solid #0C090A" }}
          >
            {updateId ? "Update task" : "Add task"}
          </button>
        </div>
        <ul className="todo-list">
          {tasks.map((task) => (
            <List
              key={task._id}
              id={task._id}
              task={task.task}
              setUpdateUI={setUpdateUI}
              updateMode={updateMode}
            />
          ))}
        </ul>
      </div>

      {/* simple crud operation */}
      {/* <CrudOperation/> */}
      
    </>
  );
}

export default App;
