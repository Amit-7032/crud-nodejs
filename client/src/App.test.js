import React from "react";
import App from "./App";
import { shallow } from "enzyme";
import axios from "axios";

// jest.mock("axios", () => {
//   return {
//       __esModule: true,
//       axios: jest.fn(),
//   };
// });

function sum(a,b) {
    return a+b;
}

// test("check sum is working properly", () => {
//     const result = sum(2,3);
//     expect(result).toBe(5)
// })

describe("Given App", () => {
  test("Test1", () => {
    const result = sum(2,3);
    expect(result).toBe(5)
  });
  test("Test2 List is render", () => {
    const component = shallow(<App />)
    const listExist = component.find("List").exists()
    expect(listExist).toBe(false)
    expect(component).toMatchSnapshot()
  });
  test("Test3", () => {
    const component = shallow(<App />)
    // const listExist = component.find("List").exists()
    // expect(listExist).toBe(false)
    expect(component).toMatchSnapshot()
  });
});
