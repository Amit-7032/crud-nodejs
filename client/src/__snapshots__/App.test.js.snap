// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given App Test2 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      CRUD Operations
    </h1>
    <div
      className="input-container"
    >
      <input
        onChange={[Function]}
        style={
          Object {
            "border": "2px solid #0C090A",
          }
        }
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid #0C090A",
          }
        }
        type="submit"
      >
        Add task
      </button>
    </div>
    <ul
      className="todo-list"
    />
  </div>
</Fragment>
`;

exports[`Given App Test2 List is render 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      CRUD Operations
    </h1>
    <div
      className="input-container"
    >
      <input
        onChange={[Function]}
        style={
          Object {
            "border": "2px solid #0C090A",
          }
        }
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid #0C090A",
          }
        }
        type="submit"
      >
        Add task
      </button>
    </div>
    <ul
      className="todo-list"
    />
  </div>
</Fragment>
`;

exports[`Given App Test3 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      CRUD Operations
    </h1>
    <div
      className="input-container"
    >
      <input
        onChange={[Function]}
        style={
          Object {
            "border": "2px solid #0C090A",
          }
        }
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid #0C090A",
          }
        }
        type="submit"
      >
        Add task
      </button>
    </div>
    <ul
      className="todo-list"
    />
  </div>
</Fragment>
`;
